lib/src/config/config.rs

```diff
@@ -119,7 +119,11 @@ impl Config {
                 return Err(self.bad_type(name, val, "an unsigned integer"));
             }
 
-            self.port = port as usize;
+            if port > (u16::max_value() as i64) {
```

Comment:
@SergioBenitez now that you mention it I totally see the problem. Kind of makes this unavoidable then.